src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesignerAssembly.cs

```diff
@@ -0,0 +1,352 @@
+			FieldAttributes attributes = FieldAttributes.Public, TypeAttributes typeAttributes = TypeAttributes.NestedPublic)
+		{
+			TypeDefinition nestedType = CreateResourceClass (resourceDesigner, resourceClass, module, typeAttributes);
+			FieldDefinition p = CreateField (fieldName, value, module, attributes);
+			nestedType.Fields.Add (p);
+		}
+
+		void CreateIntArrayProperty (string resourceClass, string propertyName, int[] values, TypeDefinition resourceDesigner, ModuleDefinition module,
+			MethodAttributes attributes = MethodAttributes.Public, TypeAttributes typeAttributes = TypeAttributes.NestedPublic)
+		{
+			TypeDefinition nestedType = CreateResourceClass (resourceDesigner, resourceClass, module, typeAttributes);
+			PropertyDefinition p = CreateArrayProperty (propertyName, values, module, attributes);
+			nestedType.Properties.Add (p);
+			nestedType.Methods.Insert (Math.Max(0, nestedType.Methods.Count () -1), p.GetMethod);
+		}
+
+		void CreateIntArrayField (string resourceClass, string fieldName, int[] values, TypeDefinition resourceDesigner, ModuleDefinition module,
+			FieldAttributes attributes = FieldAttributes.Public, TypeAttributes typeAttributes = TypeAttributes.NestedPublic)
+		{
+			TypeDefinition nestedType = CreateResourceClass (resourceDesigner, resourceClass, module, typeAttributes);
+			FieldDefinition p = CreateArrayField (fieldName, values, module, attributes);
+			nestedType.Fields.Add (p);
+			MethodDefinition ctor = CreateStaticCtor (nestedType, module);
+			ILProcessor il = ctor.Body.GetILProcessor ();
+			il.Emit(OpCodes.Ldc_I4, values.Count ()); // store array size
+			il.Emit(OpCodes.Newarr, intRef); //create a new  array
+			il.Emit(OpCodes.Stsfld, p);
+			int index = 0;
+			foreach (int value in values) {
+				//il.Emit (OpCodes.Dup);
+				il.Emit (OpCodes.Ldsfld, p);
+				il.Emit (OpCodes.Ldc_I4, index++); // index
+				il.Emit (OpCodes.Ldc_I4, value); // value
+				il.Emit (OpCodes.Stelem_I4);
+			}
+			//il.Emit (OpCodes.Ret);
+		}
+
+		Dictionary<string, TypeDefinition> resourceClasses = new Dictionary<string, TypeDefinition> (StringComparer.OrdinalIgnoreCase);
+		Dictionary<string, MethodDefinition> staticConstructors = new Dictionary<string, MethodDefinition> ();
+
+		void CreateCtor (TypeDefinition type, ModuleDefinition module)
+		{
+			var ctor = new MethodDefinition(".ctor", MethodAttributes.SpecialName | MethodAttributes.RTSpecialName | MethodAttributes.HideBySig | MethodAttributes.Public, module.TypeSystem.Void);
+			var ctoril = ctor.Body.GetILProcessor();
+			ctoril.Emit(OpCodes.Ldarg_0);
+			var o = module.TypeSystem.Object.Resolve();
+			ctoril.Emit(OpCodes.Call, module.ImportReference(o.Methods.First(x => x.IsConstructor)));
+			ctoril.Emit(OpCodes.Ret);
+			type.Methods.Add(ctor);
+		}
+
+		MethodDefinition CreateStaticCtor (TypeDefinition type, ModuleDefinition module)
```

Comment:
Since this method does not always create a `.cctor`, this should probably be named `GetStaticCtor()`.